File: practice16/8.cpp
//https://www.nowcoder.com/practice/a5edebf0622045468436c74c3a34240f?tpId=40&tqId=21349&tPage=1&rp=1&ru=/ta/kaoyan&qru=/ta/kaoyan/question-ranking
#include<iostream>
#include<string>
using namespace std;
int main()
{
	string s1, s2;
	int i, j, m, n, sum, x, y;
	while (cin >> s1 >> s2)
	{
		m = s1.size();
		n = s2.size();
		sum = 0;
		for (i = 0; i < m; i++)
		{
			x = s1[i] - '0';
			for (j = 0; j < n; j++)
			{
				y = s2[j] - '0';
				sum = sum + x*y;
			}
		}
		cout << sum << endl;
	}
	system("pause");
	return 0;
}

File: practice29/8.java


//https://www.nowcoder.com/practice/07bca1eb592146eab534bdfc099db7f5?tpId=98&tqId=32918&tPage=5&rp=5&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.io.*;

public class Main{
    public static void main(String []args) throws IOException {
        BufferedReader bufferedReader=new BufferedReader(new InputStreamReader(System.in));
        String s=bufferedReader.readLine();
        Sum(s);
    }
    public static void Sum(String s){
        String []str=s.split(",");
        int value=Integer.parseInt(str[str.length-1]);
        int sum=0;
        for(int i=0;i<str.length-1;i++){
            for(int j=i+1;j<str.length-1;j++){
                if(Integer.parseInt(str[i])+Integer.parseInt(str[j])==value){
                    sum=sum+j+i;
                }
            }
        }
        System.out.println(sum);
    }
}
